Core/src/org/sleuthkit/autopsy/casemodule/Case.java

```diff
@@ -583,24 +583,41 @@ public static boolean isCaseOpen() {
     }
 
     /**
+     * Deprecated. Use getOpenCase() instead.
+     * 
      * Gets the current case, if there is one, at the time of the call.
      *
      * @return The current case.
      *
      * @throws IllegalStateException if there is no current case.
      */
+    @Deprecated
     public static Case getCurrentCase() {
         /*
          * Throwing an unchecked exception is a bad idea here.
          *
-         * TODO (JIRA-2229): Case.getCurrentCase() method throws unchecked
-         * IllegalStateException; change to throw checked exception or return
-         * null
          */
-        if (null != currentCase) {
-            return currentCase;
+        try {
+            Case curCase = getOpenCase();
+            return curCase;
+        } catch (NoCurrentCaseException e) {
+            throw new IllegalStateException(NbBundle.getMessage(Case.class, "Case.getCurCase.exception.noneOpen"), e);
+        }
+    }
+
+    /**
+     * Gets the current open case, if there is one, at the time of the call.
+     *
+     * @return The open case.
+     *
+     * @throws NoCurrentCaseException if there is no open case.
+     */
+    public static Case getOpenCase() throws NoCurrentCaseException {
+        Case openCase = currentCase;
+        if (null != openCase) {
+            return openCase;
         } else {
-            throw new IllegalStateException(NbBundle.getMessage(Case.class, "Case.getCurCase.exception.noneOpen"));
+            throw new NoCurrentCaseException();
```

Comment:
It is not a good idea to throw an exception with no message, e.g., it is awkward when exceptions are logged. See the comment below on the constructors this exception class should have.